codalab/apps/api/views/storage_views.py

```diff
@@ -0,0 +1,116 @@
+import datetime
+
+from django.conf import settings
+from django.utils.dateparse import parse_datetime
+
+from rest_framework import (permissions, status, viewsets, views, filters, mixins)
+from rest_framework.decorators import action, link, permission_classes
+from rest_framework.exceptions import PermissionDenied, ParseError
+from rest_framework.response import Response
+
+from apps.authenz.models import ClUser
+from apps.web.models import Competition
+from apps.web.utils import get_competition_size_data, BundleStorage, storage_recursive_find, storage_get_total_use
+
+
+@permission_classes((permissions.IsAuthenticated,))
+class GetStorageAnalyticTotalsView(views.APIView):
+    """
+    Gets total usage of bucket by looping through each key and adding the total size
+    """
+    def get(self, request, *args, **kwargs):
+        if not self.request.user.is_staff:
+            raise PermissionDenied(detail="Admin only")
+        total_bytes = storage_get_total_use(BundleStorage)
+        data = {
+            'total_bytes': total_bytes,
+            'total_kilobytes': '{:.2f}'.format(total_bytes / 1000),
+            'total_megabytes': '{:.2f}'.format(total_bytes / 1000 / 1000),
+            'total_gigabytes': '{:.2f}'.format(total_bytes / 1000 / 1000 / 1000),
+        }
+        resp_status = status.HTTP_200_OK
+        return Response(data, status=resp_status)
+
+
+@permission_classes((permissions.IsAuthenticated,))
+class GetCompetitionStorageAnalytics(views.APIView):
+    """
+    Gets storage use for competitions
+    """
+    def get(self, request, *args, **kwargs):
+        if not self.request.user.is_staff:
+            raise PermissionDenied(detail="Admin only")
+
+        comp_list = []
+        qs = Competition.objects.all().reverse()
+        for comp in qs:
+            comp_list.append(get_competition_size_data(comp))
+
+        return Response(comp_list, status=status.HTTP_200_OK)
+
+
+@permission_classes((permissions.IsAuthenticated,))
+class GetUserStorageAnalytics(views.APIView):
+    """
+    Gets storage use for users
+    """
+    def get(self, request, *args, **kwargs):
+        if not self.request.user.is_staff:
+            raise PermissionDenied(detail="Admin only")
+
+        user_list = []
+        qs = ClUser.objects.order_by('?').exclude(id=-1)
+        for user in qs:
+            user_list.append(user.get_storage_use_data())
```

Comment:
isn't this insanely slow? can't  you do some kind of batch command or something?

To do better analytics, maybe you could just run a script on the minio server and that would return like instantly?

This would almost ddos minio ?